src/main/java/globalquake/ui/stationselect/DeselectAllAction.java

```diff
@@ -19,11 +19,21 @@ public DeselectAllAction(StationDatabaseManager stationDatabaseManager) {
 
     @Override
     public void actionPerformed(ActionEvent actionEvent) {
+        boolean alreadyDeselected = true;
         stationDatabaseManager.getStationDatabase().getDatabaseWriteLock().lock();
         try{
             for(Network network : stationDatabaseManager.getStationDatabase().getNetworks()){
+                for(globalquake.database.Station station : network.getStations()){
+                    if(station.getSelectedChannel() != null){
+                        alreadyDeselected = false;
+                        break;
+                    }
+                }
                 network.getStations().forEach(station -> station.setSelectedChannel(null));
             }
+            if(alreadyDeselected){
+                JOptionPane.showMessageDialog(null, "All Stations Already Deselected");
```

Comment:
Try to pass the value of StationSelectFrame here so that it can be inserted as the first parameter of the
`showMessageDialog()` method. This will make the popup dialog appear centered to the main frame. This is particularly important for multi-monitor setups, where the dialog needs to appear on the correct monitor.